specs/executionAPISpec.md

```diff
@@ -0,0 +1,530 @@
+## <p id="hashrate">eth_hashrate</p>
+* [EH-1] eth_hashrate **MUST** return the hashes per second that the client is using to mine blocks.
+* [EH-2] eth_hashrate **MUST** return 0x0 when the client does not have mining enabled.
+## <p id="submitHashrate">eth_submitHashrate</p>
+* [ESH-1] eth_submitHashrate **MUST** return true when the client successfully submits a `hashrate` and an `id`.
+* [ESH-2] eth_submitHashrate **MUST** return true when the client submits their hashrate while not mining.
+* [ESH-3] eth_submitHashrate **MUST** return true when the client submits their hashrate while syncing to the network.
+* [ESH-4] eth_submitHashrate **MUST** submit 0x0 when `hashrate` is null.
+* [ESH-5] eth_submitHashrate **MUST NOT** error when `id` is equal to the `id` or another mining client.
+## <p id="gasPrice">eth_gasPrice</p>
+* [EGP-1] eth_gasPrice **MUST** return the current price per unit of gas in wei that the client is charging.
+## <p id="maxPriorityFeePerGas">eth_maxPriorityFeePerGas</p>
+* [EMPFPG-1] eth_maxPriorityFeePerGas **MUST** return the clients price per unit of gas - 7.
+## <p id="feeHistory">eth_feeHistory</p>
+* [EFH-1] eth_feeHistory **MUST** return the following information for number of blocks specified by the `blockCount` parameter stopping at the `highestBlock` parameter.
+  * An array containing the base fee per gas for each block plus then block after `highestBlock`.
+  * An array containing the ratio of the gas used by each block.
+  * An array containing arrays with the requested `rewardPercentiles` for each block.
+  * The oldest block used for the request.
+* [EFH-2] eth_feeHistory **MUST** use the available range of blocks when the requested `blockCount` range can't be retrieved.
+* [EFH-3] eth_feeHistory **MUST** allow block tags to be used for `highestBlock`.
+* [EFH-4] eth_feeHistory **MUST** error with code -32000 when `highestBlock` is ahead of the chain.
+## <p id="blockNumber">eth_blockNumber</p>
+* [EBN-1] eth_blockNumber **MUST** return the block number for the most recent block mined.
+* [EBN-2] eth_blockNumber **MUST** return "0x0" when the client is not synced to the network.
+## <p id="sign">eth_sign</p>
+* [ESN-1] eth_sign **MUST** return the Ethereum specific signature detailed in [EIP-191](https://eips.ethereum.org/EIPS/eip-191) for the given unlocked `address` and `message`.
+* [ESN-2] eth_sign **MUST** error with code -32000 when the account corresponding to the `address` is not unlocked.
+* [ESN-3] eth_sign **MUST** error with code -32000 when the account corresponding to the `address` is not owned by the client.
+## <p id="call">eth_call</p>
+* [EC-1] eth_call **MUST** return the result of the given transaction.
+* [EC-2] eth_call **MUST** accept all current transaction types. Legacy transactions and [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) "typed" transactions.
```

Comment:
```suggestion
 * [EC-2] eth_call **MUST** accept all transaction types valid for the block that the call is being executed against. This may include legacy transactions or [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) "typed" transactions.
```